### DISCUSSION TITLE
Provide persistent logs in a specified path

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#471] [Feedback & Feature Proposal]
Following this issue: https://github.com/meilisearch/meilisearch/issues/898

Currently the users have to redirect the logs of Meilisearch into a single file, which is not really convenient

This discussion here is to discuss what would be the best solution for the users wanting to keep the Meilisearch logs. Please feel free to share your use-case and needs 😄

**Comment by kuchaguangjie** [comment]
One possible solution is `log rotation`, with options:

- (a). max size of each log file, (e.g 50Mb)
- (b). max duration of each log file, (e.g 7 days)
- (c). max log files to keep (e.g 20)

BTW:
- `a` and `b`, may be designed as exclusive, or co-existing.
- When `c` is 1 or 2, it basically equals to keep only 1 (or 2) log file, but with size (or duration) limit.

Relevant api & code:
- https://docs.rs/file-rotate/latest/file_rotate/
- https://github.com/emabee/flexi_logger

**Comment by Kerollmops** [comment]
Hey,

I respond here about the fact that I am not sure that it is our job to manage the logs, a lot of other tools handle that very nicely. For example, when you set up a service in systemd they automatically aggregate the stdout [with journald which kind of compresses the logs of the program](https://en.wikipedia.org/wiki/Systemd#Ancillary_components) and, tarballs them. Journald can also be used to fetch the logs by date and other settings.

There is plenty of solution to manage the logs and it is another problem that I am not sure we want to tackle by ourselves.

**Reply by gmourier** [reply]
@kuchaguangjie Why do you need a log file written by Meilisearch?

My current hypothesis is that not all users are experts and are not necessarily comfortable with journald etc.

**Reply by kuchaguangjie** [reply]
First, without a log file, there is no where to check for what happened.
Second, if the feature can be provided, then every user can save their time, no matter is it simple or difficult.

**Reply by michaelklopf** [reply]
It's hard to find the problem when no tasks are created, no log can easily be found (there is one in docker, but it shows nothing), all while the requests we send don't fail. A little hard to debug problems without an easy findable log that has hopefully logged every problem.

**Comment by curquiza** [comment]
A suggestion of workaround here: https://github.com/meilisearch/meilisearch/issues/898#issuecomment-1602975353
